tests/test-mongoc-collection.c

```diff
@@ -648,13 +697,22 @@ test_index_storage (void)
    bson_error_t error;
    bool r;
    bson_t keys;
-
-   mongoc_index_opt_init (&opt);
-   mongoc_index_opt_wt_init (&wt_opt);
+   const char *engine;
 
    client = mongoc_client_new (gTestUri);
    ASSERT (client);
 
+   /* Skip unless we are on WT */
+   engine = storage_engine(client);
+   if (strcmp(engine, "wiredtiger") != 0 &&
```

Comment:
Id be more concerned that storage_engine() will absolutely always return you non-NULL (so you don't crash strcmp). Idiomatically, it really depends who you are talking to. I find the != show more clearly that you definitely didin't accidentally add a strcmp() and forgot that it was 0 based. Or my personal preference, !!strcmp(..).

Either way, I'd probably suggest a NULL check.